sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -592,6 +592,19 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
     }
   }
 
+
+  /**
+   * An extractor that matches all binary comparison operators except null-safe equality.
+   *
+   * null-safe equality is not supported by Hive metastore partition predicate pushdown
+   */
+  object OperatorsInMetastorePartitionFPD {
```

Comment:
This name looks weird... maybe just `SpecialBinaryComparison` and explain the detail in the document.

Reply:
Done.